examples/models/ResNet50.cpp

```diff
@@ -0,0 +1,1293 @@
+        primitive_attr inner_product_attr;
+        inner_product_attr.set_post_ops(inner_product_ops);
+        inner_product_pd = inner_product_forward::primitive_desc(
+                inner_product_d, inner_product_attr, eng);
+    } else {
+        inner_product_pd
+                = inner_product_forward::primitive_desc(inner_product_d, eng);
+    }
+
+    auto inner_product_prim = inner_product_forward(inner_product_pd);
+    net.push_back(inner_product_prim);
+    net_args.push_back(
+            {{DNNL_ARG_SRC, src_mem}, {DNNL_ARG_WEIGHTS, weights_mem},
+                    {DNNL_ARG_BIAS, bias_mem}, {DNNL_ARG_DST, dst_mem}});
+}
+
+void do_binary(engine &eng, std::vector<primitive> &net,
+        std::vector<std::unordered_map<int, memory>> &net_args,
+        memory &src0_mem, memory &src1_mem, memory &dst_mem) {
+    auto src0_md = src0_mem.get_desc();
+    auto src1_md = src1_mem.get_desc();
+    auto dst_md = dst_mem.get_desc();
+    auto binary_desc
+            = binary::desc(algorithm::binary_add, src0_md, src1_md, dst_md);
+    auto binary_prim_desc = binary::primitive_desc(binary_desc, eng);
+    auto binary_prim = binary(binary_prim_desc);
+    net.push_back(binary_prim);
+    net_args.push_back({{DNNL_ARG_SRC_0, src0_mem}, {DNNL_ARG_SRC_1, src1_mem},
+            {DNNL_ARG_DST, dst_mem}});
+}
+
+struct block_params {
+
+    block_params(int layer_id, int block_id, int branch_id, int branch_block_id,
+            bool with_relu, bool with_bias, memory::dim batch, memory::dim ic,
+            memory::dim oc, memory::dim oh, memory::dim ow, memory::dim kh,
+            memory::dim kw, memory::dims padding, memory::dims strides)
+        : layer_id(layer_id)
+        , block_id(block_id)
+        , branch_id(branch_id)
+        , branch_block_id(branch_block_id)
+        , with_relu(with_relu)
+        , with_bias(with_bias)
+        , batch(batch)
+        , ic(ic)
+        , oc(oc)
+        , oh(oh)
+        , ow(ow)
+        , kh(kh)
+        , kw(kw)
+        , padding(padding)
+        , strides(strides) {}
+    int layer_id;
+    int block_id;
+    int branch_id;
+    int branch_block_id;
+    bool with_relu;
+    bool with_bias;
+    memory::dim batch;
+    memory::dim ic;
+    memory::dim oc;
+    memory::dim oh, ow;
+    memory::dim kh, kw;
+    memory::dims padding;
+    memory::dims strides;
+};
+class res_net_block {
+public:
+    res_net_block(engine &eng, std::vector<primitive> &net,
+            std::vector<std::unordered_map<int, memory>> &net_args,
+            memory &src_mem, block_params params)
+        : eng_(eng)
+        , net_(net)
+        , net_args_(net_args)
+        , src_mem_(src_mem)
+        , params_(params) {
+
+        dst_dims_ = {params.batch, params.oc, params.oh, params.ow};
+        weights_dims_ = {params.oc, params.ic, params.kh, params.kw};
+        if (params.with_bias) bias_dims_ = {params.oc};
+        // auto src_md_nchw
```

Comment:
Is it intentional to keep commented out code here?

Reply:
Removed.